### PR TITLE
Add `testfilter` subcommand for build pipelines

### LINKED ISSUE #4: Add dual-output for compatibility with existing tools
Many tools currently parse the output of `go test`, which means we can't just drop this helpful tool into a build pipeline without breaking things.

Of note, there's a proposal in upstream Golang to support a standard "go test" output (https://github.com/golang/go/issues/2981) which would help this and many other tools. However this is just a "proposal" and there's no concrete timeline for implementing.

In the meantime, the output of "go test" is the canonical source-of-truth for test runs, and many tools (including `go-junit-report`(https://github.com/jstemmer/go-junit-report) which I use and contribute to expect the tests runs to output their results in a specific, human-friendly format.

In order to differentiate between output made for humans (which richgo does a great job at doing) and output for machines (which `go test` currently does, although it's currently both for-machine and for-human, and it isn't very great for humans), I'd like to augment `richgo` to support outputting both: a human-friendly mode, and a machine-friendly mode.

In the short term, the machine-friendly mode will simply be the input we get from running "go test" unmunged. In the future, it will probably be `go test -json` or whatever the upstream proposal lands on, which will make everybody's job easier.

The primary changes proposed are to implement two flags (and optionally environment variables): 

- the output path to send the human-friendly version to (default: `os.Stdout`) 
- the output path to send the machine-friendly version to (currently `ioutil.Discard`)

note that rather than hard-coding paths for defaults, will need to do some portability things to make sure we don't hardcode this as `/dev/stdout` and instead use the more portable `os.Stdout` and `ioutil.Discard` istead of `/dev/null` -- none of this will be difficult but writing it down here so I don't forget it.

Then somebody could run this tool using linux FIFOS (totally untested): 

```bash
tmpdir=$(mkdtemp -d)
outfifo="${tmpdir}/MYFIFO
mkfifo a=rw "${outfifo}"
go-junit-report < ${outfifo} > junit.xml
richgo test --machine-friendly-output "${outfifo}"
```

And on their console they would have the default, nice, colorized output on `os.Stdout`, but `go-junit-report` would be parsing the "machine-friendly" version.

References #3

### PR DESCRIPTION
**Pull request by sectioneight**
Fixes #4 

See readme and issue for further explanation

### CODE REVIEW

**Comment by sectioneight** [comment]
Proof of concept only. If this is the approach you were thinking of, I'll add tests and clean stuff up.

File: richgo.go [hunk 1]
```diff
@@ -30,27 +32,25 @@ func main() {
...
 	default:
-		cmd = exec.Command("go", os.Args[1:]...)
-		// select a wrapper with subcommand
-		if f, ok := lps[os.Args[1]]; ok {
-			factory = f
+		// This is a bit of a special case. Somebody is already
+		// running `go test` for us, and just wants us to prettify the
+		// output.
+		if os.Args[1] == testFilterCmd {
+			cmd = exec.Command("cat", "-")
```
Comment (sectioneight) [inline richgo.go]: On the fence as to whether this is hacky or elegant. It requires the least code change, but feels a bit silly to spawn a `cat` process just to read from stdin. Then again, it fits naturally with the existing code...

File: richgo.go [hunk 2]
```diff
@@ -30,27 +32,25 @@ func main() {
...
 		}
 	}
 
-	var stderr io.WriteCloser
-	if editor.Formattable(os.Stderr) {
-		stderr = editor.Stream(os.Stderr, factory())
-	} else {
-		stderr = editor.Stream(os.Stderr, editor.Parrot())
-	}
+	stderr := formatWriteCloser(os.Stderr, factory)
```
Comment (sectioneight) [inline richgo.go]: Let me know if you'd prefer I split this up in a separate PR. I saw it and felt compelled to de-dupe it, hope you don't mind.

File: richgo.go (hunk 1 above)
Comment (kyoh86) [inline richgo.go]: Like you say, it's a bit silly. But I like it.

File: richgo.go (hunk 2 above)
Comment (kyoh86) [inline richgo.go]: Thanks.

**Comment by kyoh86** [comment]
It is already functioning. So I merged it.
I'm adding test and refactoring them now.
Thank you very much!

**Comment by sectioneight** [comment]
:) And thank you for this awesome tool. Makes me and my teams very happy.

**Pull request merged**